### PR TITLE
Fix #2299: Improve error handling of RejectedExecutionException from ExecutorService

### LINKED ISSUE #2299: RejectedExecutionException from LogWatchCallback
I got this exception with 4.9.0 on che.openshift.io
```
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@6dc25c5c rejected from java.util.concurrent.ThreadPoolExecutor@60b9cac4[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1379)
	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
	at java.util.concurrent.Executors$DelegatedExecutorService.submit(Executors.java:678)
	at io.fabric8.kubernetes.client.dsl.internal.LogWatchCallback.onResponse(LogWatchCallback.java:158)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:203)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```
I can understand how important it is. 
Can someone help to explain what does it mean?

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Fix #2299 

## Type of change

 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java [hunk 1]
```diff
@@ -205,11 +206,15 @@ public void onOpen(WebSocket webSocket, Response response) {
...
+            if (!executorService.isShutdown()) {
+              executorService.submit(pumper);
+              started.set(true);
+              queue.add(true);
+            }
         } catch (IOException e) {
-            queue.add(new KubernetesClientException(OperationSupport.createStatus(response)));
+          queue.add(new KubernetesClientException(OperationSupport.createStatus(response)));
+        } catch (RejectedExecutionException rejectedTaskException) {
+          throw new KubernetesClientException("Unable to write to OutputStream since close() has been called.", rejectedTaskException.getCause());
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java:217]: Not really sure if this Exception we'll be ever reached (at least for the cause described in the message). The `exectorService.isShutdown()`  check should prevent the exception for the alleged causes described in the message.

Comment (rohanKanojia) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java:217]: ohk. I'll remove this catch block

**Comment by rohanKanojia** [comment]
retest this please

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**